sdk/src/main/java/io/opentelemetry/sdk/trace/SpanBuilderSdk.java

```diff
@@ -126,25 +148,114 @@
 
   @Override
   public Span startSpan() {
-    // TODO: get remoteParent span from the context if noParent=false and remoteParent/remoteParents
-    // are null
-    // TODO: correctly implement this.
-    SpanContext context =
-        SpanContext.create(
-            TraceId.fromBytes(new byte[] {1, 2, 3, 4, 5, 6, 7, 8, 8, 7, 6, 5, 4, 3, 2, 1}, 0),
-            SpanId.fromBytes(new byte[] {1, 2, 3, 4, 5, 6, 7, 8}, 0),
-            TraceOptions.builder().setIsSampled(true).build(),
-            Tracestate.getDefault());
+    Span currentSpan = null;
+    if (parentType == ParentType.CURRENT_SPAN) {
+      currentSpan = ContextUtils.getValue();
+    }
+    SpanContext parentContext = parent(parentType, currentSpan, parent, remoteParent);
+    SpanContext spanContext = createSpanContext(spanName, parentContext, sampler);
+    boolean recordEvents =
+        this.recordEvents != null ? this.recordEvents : spanContext.getTraceOptions().isSampled();
+    // TODO return DefaultSpan if not recording events
+    TimestampConverter timestampConverter = getTimestampConverter(parent);
     return RecordEventsReadableSpanImpl.startSpan(
-        context,
+        spanContext,
         spanName,
         spanKind,
-        null,
+        parentContext != null ? parentContext.getSpanId() : null,
         traceConfig,
         spanProcessor,
-        null,
-        MillisClock.getInstance(),
-        Resource.getEmpty());
+        timestampConverter,
+        clock,
+        resource);
+  }
+
+  private SpanContext createSpanContext(String name, SpanContext parentContext, Sampler sampler) {
+    TraceId traceId;
+    SpanId spanId = generateRandomSpanId(random);
+    Tracestate tracestate = Tracestate.getDefault();
+    Boolean hasRemoteParent = false;
+    if (parentContext == null || !parentContext.isValid()) {
+      // New root span.
+      traceId = generateRandomTraceId(random);
+      // This is a root span so no remote or local parent.
+      hasRemoteParent = null;
+    } else {
+      // New child span.
+      traceId = parentContext.getTraceId();
+      tracestate = parentContext.getTracestate();
+    }
+    Decision samplingDecision =
+        sampler.shouldSample(
+            parentContext,
+            hasRemoteParent,
```

Comment:
I think we should remove this for the moment because no clear use for this. I think what we should add support later is a notion of "untrusted parent". But probably that needs a RFC. So I would suggest we file an issue on the specification to remove this.

Reply:
I have passed false there for now. Based on the discussion we can remove it fro `shouldSample` method.

Reply:
Spec issue https://github.com/open-telemetry/opentelemetry-specification/issues/164